lightning/src/ln/channelmanager.rs

```diff
@@ -6058,6 +6248,28 @@ where
 				});
 				!htlcs.is_empty() // Only retain this entry if htlcs has at least one entry.
 			});
+
+			let mut intercepted_htlcs = self.pending_intercepted_htlcs.lock().unwrap();
+			intercepted_htlcs.retain(|_, htlc| {
+				if height >= htlc.forward_info.outgoing_cltv_value - HTLC_FAIL_BACK_BUFFER {
+					let prev_hop_data = HTLCSource::PreviousHopData(HTLCPreviousHopData {
+						short_channel_id: htlc.prev_short_channel_id,
+						htlc_id: htlc.prev_htlc_id,
+						incoming_packet_shared_secret: htlc.forward_info.incoming_shared_secret,
+						phantom_shared_secret: None,
+						outpoint: htlc.prev_funding_outpoint,
+					});
+
+					let requested_forward_scid /* intercept scid */ = match htlc.forward_info.routing {
+						PendingHTLCRouting::Forward { short_channel_id, .. } => short_channel_id,
+						_ => unreachable!(),
+					};
+					timed_out_htlcs.push((prev_hop_data, htlc.forward_info.payment_hash,
+							HTLCFailReason::Reason { failure_code: 0x1000 | 14, data: Vec::new() },
```

Comment:
Here can we do `0x1000 | 10` as in `fail_intercepted_htlc()`, iirc we can allow outbound forward of HTLC up to 2016 blocks on the outgoing CLTV, so a HTLC can be a) an intercepted one and b) hit our `HTLC_FAIL_BACK_BUFFER` and actually still not be `expiry_too_soon`, I think ?

Reply:
Hmm, I still think this error makes more sense. See this discussion: https://github.com/lightningdevkit/rust-lightning/pull/1835#discussion_r1024440347 
For this current error, the spec says "The CLTV expiry is too close to the current block height for safe handling by the processing node." So I don't think specific block heights are prescribed, it's just based on the opinion of the processing node

Reply:
Note that we'll currently return `UPDATE|14` for phantom payment errors while leaving the channel update as `None`, arguably could fix both of them in follow-up?

Reply:
I guess it would make more sense to fix this one now, though. So basically just pointing out there may be another place to fix this

Reply:
Updated to use `temporary_node_failure`